src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] !== 'undefined' && this[i] !== null) {
```

Comment:
```suggestion
      if (char !== undefined && char !== null) {
```
let's make it work like this

Reply:
I have changed remarks, but i didnt use 'for of' loop, becouse i used index later, when i pointed to last element of array